packages/tools/src/tools/segmentation/SegmentSelectTool.ts

```diff
@@ -106,75 +116,28 @@ class SegmentSelectTool extends BaseTool {
       return;
     }
 
-    const labelmapData = activeSegmentation.representationData.LABELMAP;
-
-    let hoveredSegmentIndex;
+    const { segmentationId, segmentationRepresentationUID } =
+      activeSegmentationReps;
 
-    if (isVolumeSegmentation(activeSegmentation.representationData.LABELMAP)) {
-      const { volumeId } = labelmapData as LabelmapSegmentationDataVolume;
-
-      const segmentationVolume = cache.getVolume(volumeId);
-
-      if (!segmentationVolume) {
-        return;
-      }
-
-      hoveredSegmentIndex =
-        segmentationVolume.imageData.getScalarValueFromWorld(worldPoint);
-    } else {
-      const { imageIdReferenceMap } =
-        labelmapData as LabelmapSegmentationDataStack;
-
-      const currentImageId = viewport.getCurrentImageId();
-      const segmentationImageId = imageIdReferenceMap.get(currentImageId);
-
-      const image = cache.getImage(segmentationImageId);
-
-      if (!image) {
-        return;
-      }
-
-      const activeSegmentationRepresentation =
-        getActiveSegmentationRepresentation(this.toolGroupId);
-
-      if (!activeSegmentationRepresentation) {
-        return;
-      }
-
-      const segmentationActor = viewport.getActor(
-        activeSegmentationRepresentation.segmentationRepresentationUID
-      );
-
-      const imageData = segmentationActor?.actor.getMapper().getInputData();
-
-      const indexIJK = csUtils.transformWorldToIndex(imageData, worldPoint);
-
-      // since it is a stack we don't need to check the z
-      const flattenedIndex = indexIJK[0] + indexIJK[1] * image.columns;
-
-      const scalars = imageData.getPointData().getScalars().getData();
-
-      hoveredSegmentIndex = scalars[flattenedIndex];
-    }
+    const hoveredSegmentIndex = getSegmentAtWorldPoint(
```

Comment:
What is the impact of this hover test call when there are a lot of segments?  Although it is nice to do on hover, I do wonder about the total cost of this and impact on interaction.

Reply:
This will test against only one representation UID (active one), so it will not have performance degradation.